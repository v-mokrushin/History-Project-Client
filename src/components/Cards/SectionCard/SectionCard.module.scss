@import "/src/assets/css/common.scss";

.root {
  width: 100%;
  height: 220px;
  position: relative;
  text-decoration: none;
  transition: transform 300ms ease;
  overflow: hidden;
  display: flex;
  align-items: flex-end;
  box-shadow: 0px 0px 30px 0px rgba(36, 39, 42, 0.3);
  transition: height 300ms ease;
  background-color: #000000;

  @include breakpoint(xs) {
    max-height: 170px;
  }

  &::after {
    position: absolute;
    content: "";
    background-color: rgba($color: #000000, $alpha: 0.35);
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;
    transition: background-color 300ms ease;
  }

  &:hover::after {
    background-color: rgba($color: #000000, $alpha: 0.45);
  }

  &:hover .title {
    color: $gold;
    transform: translateX(10%);
  }

  &:hover .backgroundWrapper {
    transform: scale(1.075);
  }
}

.backgroundWrapper {
  position: absolute;
  height: 100%;
  width: 100%;
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  transition: transform 300ms ease;
  z-index: 0;
  will-change: transform;
}

.contentWrapper {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  justify-content: flex-start;
  gap: 15px;
  position: relative;
  padding: 20px 30px;
  z-index: 3;

  .title {
    font-family: "Oswald", sans-serif;
    font-weight: 400;
    font-size: 35px;
    letter-spacing: 1px;
    color: $white;
    transition: transform 300ms ease, color 300ms ease;

    @include breakpoint(xxs) {
      font-size: 20px;
    }
  }
}
