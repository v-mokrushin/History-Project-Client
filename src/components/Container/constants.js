export const CONTAINER_TYPES = {
  default: "default",
  article: "article",
};
