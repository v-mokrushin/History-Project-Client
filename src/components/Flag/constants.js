export const FLAG_COUNTRY = {
  USSR: "soviet-union",
  germany: "germany",
  USA: "united-states",
  unitedKindom: "united-kindom",
  japan: "japan",
};
