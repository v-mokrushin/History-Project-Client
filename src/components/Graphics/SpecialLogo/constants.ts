export enum SPECIAL_LOGO_TYPE {
  loading,
  inDevelopment,
  notFound,
}
