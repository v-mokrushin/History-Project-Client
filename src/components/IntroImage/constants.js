export const INTRO_IMAGE_TYPE = {
  pages: {
    articles: "articles",
    weapons: {
      aviation: "aviation",
      armored: "armored",
      smallArms: "smallArms",
      artillery: "artillery",
    },
  },
};
