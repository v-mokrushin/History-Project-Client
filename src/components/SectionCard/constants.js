export const SECTION_CARD_TYPE = {
  articles: {
    name: "Статьи",
    path: "articles",
    backgroundImage: "/images/backgrounds/hp-bg-articles.jpg",
  },
  weapons: {
    name: "Вооружения",
    path: "weapons",
    backgroundImage: "/images/backgrounds/hp-bg-weapons.png",
  },
  battles: {
    name: "Сражения",
    path: "battles",
    backgroundImage: "/images/backgrounds/hp-bg-battles.jpg",
  },

  pages: {
    weapons: {
      aviation: {
        name: "Авиация",
        path: "aviation",
        backgroundImage: "/images/backgrounds/weapons/aviation_2.jpg",
      },
      armored: {
        name: "Бронетехника",
        path: "armored-vehicles",
        backgroundImage: "/images/backgrounds/weapons/armored.jpg",
      },
      smallArms: {
        name: "Стрелковое оружие",
        path: "small-arms",
        backgroundImage: "/images/backgrounds/weapons/gun.jpg",
      },
      artillery: {
        name: "Артиллерия",
        path: "artillery",
        backgroundImage: "/images/backgrounds/weapons/artillery.jpg",
      },
    },
  },
};
