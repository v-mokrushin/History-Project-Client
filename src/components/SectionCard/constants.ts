export interface ISectionCard {
  name: string;
  path: string;
  backgroundImage: string;
}

export const SECTION_CARD_TYPE = {
  articles: {
    name: "Статьи",
    path: "articles",
    backgroundImage: "/images/backgrounds/articles.jpg",
  },
  weapons: {
    name: "Вооружения",
    path: "weapons",
    backgroundImage: "/images/backgrounds/weapons.jpg",
  },
  battles: {
    name: "Сражения",
    path: "battles",
    backgroundImage: "/images/backgrounds/battles.jpg",
  },

  pages: {
    weapons: {
      aviation: {
        name: "Авиация",
        path: "aviation",
        backgroundImage: "/images/backgrounds/weapons/aviation.jpg",
      },
      armored: {
        name: "Бронетехника",
        path: "armored-vehicles",
        backgroundImage: "/images/backgrounds/weapons/armored.jpg",
      },
      smallArms: {
        name: "Стрелковое оружие",
        path: "small-arms",
        backgroundImage: "/images/backgrounds/weapons/gun.jpg",
      },
      artillery: {
        name: "Артиллерия",
        path: "artillery",
        backgroundImage: "/images/backgrounds/weapons/artillery.jpg",
      },
      grenadeLaunchers: {
        name: "Гранатометы",
        path: "grenade-launchers",
        backgroundImage: "/images/backgrounds/weapons/grenade-launchers.jpg",
      },
    },
  },
};
