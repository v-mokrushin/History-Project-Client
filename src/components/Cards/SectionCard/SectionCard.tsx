import React from "react";
import { NavLink } from "react-router-dom";
import styles from "./SectionCard.module.scss";
import classNames from "classnames";
import { ANIMATIONS } from "../../../constants/animations";

import { getBackgroundImageStyleObject } from "../../../utils/common";
import { IPage } from "constants/pages";
import { Scroll } from "utils/scroll";
import Preloader from "components/Graphics/Preloader/Preloader";

interface ISectionCardProps {
  information: IPage;
  className?: string;
}

export default function SectionCard({
  information,
  className,
}: ISectionCardProps) {
  return (
    <NavLink
      to={information.path}
      className={classNames(styles.root, className, ANIMATIONS.fadeIn)}
      onClick={Scroll.toTopInstantly}
    >
      <Preloader color="white" />
      <div
        className={classNames(styles.backgroundWrapper)}
        style={getBackgroundImageStyleObject(information.introImage)}
      ></div>
      <div className={styles.contentWrapper}>
        <p className={styles.title}>{information.name.russian}</p>
      </div>
    </NavLink>
  );
}
