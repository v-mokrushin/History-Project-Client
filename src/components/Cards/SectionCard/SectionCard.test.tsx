import { render } from "@testing-library/react";
import { Pages } from "constants/pages";
import { BrowserRouter } from "react-router-dom";
import SectionCard from "./SectionCard";

describe("SectionCard", () => {
  test("SectionCard render", () => {
    const sectionCard = render(
      <BrowserRouter>
        <SectionCard information={Pages.armoredVehicles} />
      </BrowserRouter>
    );
    expect(sectionCard).toMatchSnapshot();
  });
});
