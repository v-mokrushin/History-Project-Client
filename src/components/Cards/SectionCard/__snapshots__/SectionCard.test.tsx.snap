// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionCard SectionCard render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="root animate__animated animate__fadeIn"
        href="/armored-vehicles"
      >
        <div
          class="root white"
        />
        <div
          class="backgroundWrapper"
          style="background-image: url(/images/backgrounds/weapons/armored.jpg);"
        />
        <div
          class="contentWrapper"
        >
          <p
            class="title"
          >
            Бронетехника
          </p>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="root animate__animated animate__fadeIn"
      href="/armored-vehicles"
    >
      <div
        class="root white"
      />
      <div
        class="backgroundWrapper"
        style="background-image: url(/images/backgrounds/weapons/armored.jpg);"
      />
      <div
        class="contentWrapper"
      >
        <p
          class="title"
        >
          Бронетехника
        </p>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
